### PR TITLE
Updates for PyData theme 0.14.0

### PR DESCRIPTION
**Pull request by agriyakhetarpal**
# Description

Updates for the new version of the PyData Sphinx Theme, CSS improvements, and some cleanup with the documentation builds (dependencies, external links, etc).

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [x] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [ ] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [x] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [x] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Comment by agriyakhetarpal** [comment]
There are some errors upon adding the version warning bar (the theme switcher button stops working and our versioning for dev branches is a bit different than what the configuration requires for the announcement banner to be active). As previously discussed, I think the one from Read the Docs addons would be better.

File: noxfile.py
Comment (agriyakhetarpal) [inline noxfile.py:22]: Not needed anymore after we use separate jobs instead of running tests sequentially (#3264). I added coloured outputs to GitHub Actions too

File: noxfile.py
Comment (agriyakhetarpal) [inline noxfile.py:49]: Using `session.run_always()` as an alias to `session.install` was only needed for local development for cases where `pip` did not display an output. It was and is fine to use it in CI

Comment (agriyakhetarpal) [inline noxfile.py:49]: Using `session.run_always()` as an alias to `session.install` was only needed for local development for cases where `pip` did not display a list of dependencies installed in its output. It was and is fine to use it in CI

**Comment by agriyakhetarpal** [comment]
Upon making changes to the noxfile to bring it back to its previous state, some of the older `nox` caches have started interfering with the current workflows. @Saransh-cpp, is there a chance you could delete the [current cache entries](https://github.com/pybamm-team/PyBaMM/actions/caches) and trigger another build for this branch?

File: .github/workflows/run_periodic_tests.yml
Comment (agriyakhetarpal) [inline .github/workflows/run_periodic_tests.yml:117]: Because I noticed some errors and warnings are not caught by the doctests, this can help ensure we are free of them. We could move the examples job above in the code too so that it can run in case the docs and/or the doctests are going to fail.

**Comment by agriyakhetarpal** [comment]
Oops, didn't realise I replicated the changes I made here in #3301 😄 @Saransh-cpp this is ready for merging now, the version warning bar on the PyData Theme seems to have some issues with detecting the preferred version, we should look into that some other time

**Review by Saransh-cpp — APPROVED** [review]
This looks amazing, thanks @agriyakhetarpal!

**Comment by agriyakhetarpal** [comment]
Should I try to un-suppress the `myst.xref_warnings` here or would keeping them be okay?

**Comment by Saransh-cpp** [comment]
Keeping them suppressed should be okay if they are unfixable in our case.

**Comment by agriyakhetarpal** [comment]
I can fix them here, I opened a new issue for checking other broken links and references (#3387) if someone would want to pick it up. Also, I am considering moving the doctests steps to another job instead of running them before the notebooks as I mentioned above. Doing that will simplify the workflow (allows examples to run) and reduce CI runtime by a bit (~5-6 minutes), because this is still the longest job

**Comment by agriyakhetarpal** [comment]
Okay, should be good to go now once the tests pass. After this, it looks like we will be able to achieve sub-30-minute CI times due to the changes in #3301 (will merge `develop` there after this is merged)

**Pull request merged**
